Fix for alphabetical recursive call issue

SiriusStarr opened the pull request:
This adds a test case for the bug causing #5

SiriusStarr committed:
🧪 Add test for alphabetical recursive call issue

jfmengels commented on tests/CognitiveComplexityTest.elm:
```diff
@@ -611,6 +611,41 @@ fun5 n =
+                """module A exposing (..)
+
+b = b
+
```
I don't think this compiles, so I don't know how good of a test case this is 🤔

SiriusStarr replied:
Now it does.

jfmengels replied:
👍

SiriusStarr committed:
Make test compile

SiriusStarr committed:
Update test suite to pass if bug is fixed

SiriusStarr committed:
🐛 Fix recursive calls being missed

As functions were visited in alphabetical order, they would be marked
`Done`, leading to recursion not getting checked if the call was to an
"earlier" function.

SiriusStarr commented:
As functions were visited in alphabetical order, they would be marked
`Done`, leading to recursion not getting checked if the call was to an
"earlier" function.

Fixes #5

SiriusStarr commented:
(Incidentally, you should probably bump the `elm-review` dependency before this or anything else gets released, given bugfixes, since it's still at `2.10.0`.)

jfmengels commented:
Awesome work @SiriusStarr! Thank you for applying your great :female_detective: skills on this!
